### PR TITLE
`x/sync` -- Add `SyncableDB` proto server/client

### PR DESCRIPTION
**Pull request by danlaine**
## Why this should be merged

So that in tests, the sync manager can use a remote database.

## How this works

Pretty boilerplate. Server wraps a `SyncableDB` and serves it over gRPC, Client implements `SyncableDB` for use in tests.

## How this was tested

The server/client themselves aren't tested since there isn't really much to test, and this is only for use in tests.

### CODE REVIEW

File: x/sync/syncable_db.go [hunk 1]
```diff
@@ -3,11 +3,299 @@
...
+	var proof merkledb.RangeProof
+	if err := proof.UnmarshalProto(req.RangeProof); err != nil {
+		return nil, err
+	}
+
+	err := s.db.CommitRangeProof(ctx, req.StartKey, &proof)
+	return &emptypb.Empty{}, err
+}
+
+func NewSyncableDBClient(client syncpb.SyncableDBClient) *SyncableDBClient {
```
Comment (darioush) [inline x/sync/syncable_db.go:180]: should we consider making 2 files?

File: x/sync/syncable_db.go [hunk 2]
```diff
@@ -3,11 +3,299 @@
...
 	merkledb.ChangeProofer
 	merkledb.RangeProofer
 }
+
+func NewSyncableDBServer(db SyncableDB) *SyncableDBServer {
+	return &SyncableDBServer{db: db}
+}
+
+type SyncableDBServer struct {
+	syncpb.UnsafeSyncableDBServer
```
Comment (darioush) [inline x/sync/syncable_db.go:35]: is it better to avoid embedding this to ensure we implement all the methods?

**Review by darioush — APPROVED** [review]

File: x/sync/syncable_db.go (hunk 2 above)
Comment (danlaine) [reply x/sync/syncable_db.go:35]: This is required to implement the interface and what we do in other places (e.g. `rpcdb.DatabaseServer`)

File: x/sync/syncable_db.go (hunk 1 above)
Comment (danlaine) [reply x/sync/syncable_db.go]: done

**Pull request merged**